### PR TITLE
Clear NULL Registration Requirements.

### LINKED ISSUE #2998: Registration requirements are shown when they are empty
The registration requirements for old competitions are shown even when they are empty:

screenshot_2018-06-23_21-08-11
screenshot_2018-06-23_21-08-48

This was obviously already fixed in #2665, but seems to be broken again.

### PR DESCRIPTION
**Pull request by AlbertoPdRF**
Fixes #2998.

The truth is that now all past competitions have some kind of registration requirements ([example](https://www.worldcubeassociation.org/competitions/Spain2008)), so they will always be displayed, but at least we get rid of those unnecessary `#` characters.

### CODE REVIEW

**Comment by jfly** [comment]
Unfortunately, this conflicts with https://github.com/thewca/worldcubeassociation.org/pull/3265, which I just merged up. In the future, if you have two PRs that depend on each other, it's nice to mention it in your PR so reviewers know what's going on.

**Comment by AlbertoPdRF** [comment]
Sorry about that. Probably these two should have been a PR only, but I wasn't sure how to proceed.

File: WcaOnRails/db/migrate/20180825114051_clear_null_registration_requirements.rb [hunk 1]
```diff
@@ -0,0 +1,7 @@
+# frozen_string_literal: true
+
+class ClearNullRegistrationRequirements < ActiveRecord::Migration[5.2]
+  def change
+    Competition.visible.where(registration_requirements: "#").update_all(registration_requirements: nil)
```
Comment (jfly) [inline WcaOnRails/db/migrate/20180825114051_clear_null_registration_requirements.rb]: I would expect this to be `extra_registration_requirements` now that we've merged up https://github.com/thewca/worldcubeassociation.org/pull/3265.

Comment (AlbertoPdRF) [reply WcaOnRails/db/migrate/20180825114051_clear_null_registration_requirements.rb]: You are totally right!

**Comment by jonatanklosko** [comment]
It looks good now, but there's a conflict in `db/structure.sql`.

**Comment by AlbertoPdRF** [comment]
I think I have addressed the conflict now.

**Comment by jfly** [comment]
Looking good, thanks @AlbertoPdRF!

**Pull request merged**